### ISSUE TITLE
Deployment names should be unique per product

### ISSUE THREAD

**Issue by mobileoverlord** [nerves-hub/nerves_hub_web#225]
I went to go add a unique index, but, we only have deployments belonging to `firmware`. Everything we know about who actually owns that deployment is known through the `firmware_id`. I think we should add `org_id` and possibly `product_id` to the `deployments` table so we can create a unique index. @jlmcgehee21 What do you think about this?

**Comment by jlmcgehee21** [comment]
We've already removed `product_id` from deployment intentionally: https://github.com/nerves-hub/nerves_hub_web/blob/master/apps/nerves_hub_core/priv/repo/migrations/20180724162500_firmware_is_truth.exs

**Comment by mobileoverlord** [comment]
Ah right. 🤔

**Comment by jlmcgehee21** [comment]
I agree we need a way to uniquely identify deployments within the scope of a product.

**Comment by jlmcgehee21** [comment]
Circling back after a call I had with @mobileoverlord, we decided the best path forward is to add `product_id` back to the `deployments` table so that we can enforce the desired uniqueness constraint.

With deployments in particular, we have a case where a device needs to identify its corresponding deployments using its firmware uuid, but we also have a case where a user in the web/cli needs to be able to uniquely reference a deployment with the information they will know (product and deployment name)
